processing/src/main/java/org/apache/druid/segment/VirtualColumn.java

```diff
@@ -336,4 +336,12 @@ default ColumnIndexSupplier getIndexSupplier(
   {
     return NoIndexesColumnIndexSupplier.getInstance();
   }
+
+  /**
+   * Check if a virtual column is the same as some other virtual column, ignoring output name.
+   */
+  default boolean isEquivalent(VirtualColumn other)
+  {
+    return this.equals(other);
+  }
```

Comment:
stuff probably should implement hashcode + equals anyway, but I'm not really comfortable at this time changing the contract of equals and hashcode to not include the output name, since it is more of a behavioral change than I was really looking to make at this time.

Reply:
leaving it like this could still have adverse effects as if a descendant doesn't override equals it will fall back to `Object.equals` 

I think it would be better to annotate `VirtualColumn` with `@SubclassesMustOverrideEqualsAndHashCode` and fix the issues.